plugins/analytics/public/onyphe.py

```diff
@@ -0,0 +1,369 @@
+import sys
+import json
+import logging
+import requests
+from requests.utils import quote
+
+from core.analytics import OneShotAnalytics
+from core.errors import ObservableValidationError
+from core.observables import Hostname, Ip
+
+default_types = {
+    "Hostname": Hostname,
+    "Ip": Ip,
+}
+
+if sys.version_info.major == 2:
+    from urlparse import urljoin
+elif sys.version_info.major == 3:
+    from urllib.parse import urljoin
+
+
+"""
+onyphe.client
+~~~~~~~~~~~~~
+
+This module implements the Onyphe API.
+
+:copyright: (c) 2017- by Sebastien Larinier
+"""
+
+class APIError(Exception):
+    """This exception gets raised whenever a non-200 status code was returned by the Onyphe API."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return self.value
+
+class Onyphe:
+    """Wrapper around the Onyphe REST
+
+        :param key: The Onyphe API key that can be obtained from your account page (https://www.onyphe.io)
```

Comment:
Please indicate the type here (`str` I assume?) and leave the instructions as a comment in the docstring.

e.g.

```
"""Wrapper around the Onyphe REST API.

The Onyphe API key that can be obtained from your account page (https://www.onyphe.io)

    :param key: str
    :type key: str
"""